File: matrix/240.search-a-2-d-matrix-ii.py
#
# @lc app=leetcode id=240 lang=python3
#
# [240] Search a 2D Matrix II
#

# @lc code=start
class Solution:
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        if not matrix or not matrix[0]:
            return False
        m = len(matrix)
        n = len(matrix[0])
        r = 0
        c = n - 1
        while r < m and c >= 0:
            if matrix[r][c] == target:
                return True
            elif matrix[r][c] > target:
                c = c - 1
            else:
                r = r + 1
        return False


# @lc code=end

File: matrix/240.search-a-2-d-matrix-ii.go
/*
 * @lc app=leetcode id=240 lang=golang
 *
 * [240] Search a 2D Matrix II
 */

// @lc code=start
func searchMatrix(matrix [][]int, target int) bool {
	m := len(matrix)
	n := len(matrix[0])
	r := 0
	c := n - 1
	for r < m && c > 0 {
		if matrix[r][c] == target {
			return True
		}
		if matrix[r][c] > target {
			c = c - 1
		} else {
			r = r + 1
		}
	}
	return false
}

// @lc code=end

